schedule/models/events.py

```diff
@@ -59,8 +62,91 @@ def __unicode__(self):
     def get_absolute_url(self):
         return reverse('event', args=[self.id])
 
-    def get_occurrences(self, start, end, skip_booster=False):
+    def get_recent_start_validated(self):
+        """
+         Unserializes and validates self.recent_start. Returns the dictionary
+         represented by self.recent_start if all goes well, or None if there
+         is a validation error.
+         @return recent_start converted to object if it's valid given current
+         state of event; None if it doesn't exist or is invalid.
+        """
+        if self.recent_start == None:
+            return None
+
+        recent_start_data = pickle.loads(str(self.recent_start))
+
+        if (recent_start_data['event']['start'] == self.start and
+              recent_start_data['event']['rule'] == self.rule):
+            return recent_start_data
+        else:
+            return None
+
+
+    @property
+    def recent_occurrence_start(self):
         """
+        A lazy loaded property that ensures we unpickle to get the recent
+        occurrence start only once. Use similar pattern for other properties
+        :return:
+        """
+        if not hasattr(self, '__recent_start'):
```

Comment:
Best practice around cached properties is to add one `_` before the property name when defining the "private" property: `_recent_occurrence_start`